tracer/src/Datadog.Trace/ExtensionMethods/StringExtensions.cs

```diff
@@ -129,5 +129,70 @@ public static bool TryConvertToNormalizedTagName(this string value, bool normali
             normalizedTagName = StringBuilderCache.GetStringAndRelease(sb);
             return true;
         }
+
+        /// <summary>
+        /// Compact whitespaces in a string
+        /// https://github.com/DataDog/datadog-agent/blob/0454961e636342c9fbab9e561e6346ae804679a9/pkg/trace/obfuscate/obfuscate.go#L118-L146
+        /// </summary>
+        /// <param name="value">String instance</param>
+        /// <returns>Whitespace compacted string instance</returns>
+        public static string CompactWhiteSpace(this string value)
+        {
+            if (string.IsNullOrWhiteSpace(value))
+            {
+                return value;
+            }
+
+            var sb = StringBuilderCache.Acquire(value.Length);
+            for (var i = 0; i < value.Length; i++)
+            {
+                var nextWhiteSpace = value.IndexOf(' ', i);
+                if (nextWhiteSpace != -1)
+                {
+                    var count = nextWhiteSpace - i;
+                    sb.Append(value, i, count > 0 ? count + 1 : count);
+                    i = nextWhiteSpace;
+                }
+                else
+                {
+                    sb.Append(value, i, value.Length - i);
+                    i = value.Length;
+                }
+            }
+
+            return StringBuilderCache.GetStringAndRelease(sb);
+        }
+
+        /// <summary>
+        /// Replaces digits with ?
+        /// https://github.com/DataDog/datadog-agent/blob/0454961e636342c9fbab9e561e6346ae804679a9/pkg/trace/obfuscate/obfuscate.go#L148-L167
+        /// </summary>
+        /// <param name="value">String instance</param>
+        /// <returns>Filtered string instance</returns>
+        public static string ReplaceDigits(this string value)
+        {
+            bool scanningDigit = false;
+            var sb = StringBuilderCache.Acquire(value.Length);
+
+            for (var i = 0; i < value.Length; i++)
+            {
+                if (char.IsDigit(value[i]))
+                {
+                    if (scanningDigit)
+                    {
+                        continue;
+                    }
+
+                    scanningDigit = true;
+                    sb.Append('?');
+                    continue;
+                }
+
+                scanningDigit = false;
+                sb.Append(value[i]);
+            }
+
+            return StringBuilderCache.GetStringAndRelease(sb);
+        }
```

Comment:
I wonder if "cleaning" methods like this should exist in their own static class to avoid polluting the `string` namespace (IntelliSense, for example).

If we do this, we should consider also moving `UriHelper.IsIdentifierSegment()` which is similar to your `ReplaceDigits()` but also accepts hex-looking strings (`[0-9A-fa-f]`) under certain conditions.

Reply:
Resolving this comment since it's not part of your changes.